src/stream/src/from_proto/dedup.rs

```diff
@@ -45,7 +45,7 @@ impl ExecutorBuilder for AppendOnlyDedupExecutorBuilder {
             .iter()
             .map(|idx| *idx as _)
             .collect_vec();
-        Ok(Box::new(AppendOnlyDedupExecutor::new(
+        Ok(Box::new(DedupExecutor::new(
```

Comment:
I guess this can cause incompatibility? Legacy protobuf will lead to `DedupExecutor` operating on a incompatible state table.

Can we achieve this RP in a backward-compatible way? Like, leave the `AppendOnlyDedupExecutor` as it is, mark it as deprecated, rename the old `DedupNode` to `AppendOnlyDedupNode` and add a new `DedupNode` for the new stream plan node variant in case of future extension.